knora/dsplib/utils/xml_upload.py

```diff
@@ -730,35 +889,82 @@ def xml_upload(input_file: str, server: str, user: str, password: str, imgdir: s
-                                                                       iri=resource_iri,
-                                                                       permissions=permissions_tmp,
-                                                                       bitstream=resource_bitstream,
-                                                                       values=resource.get_propvals(res_iri_lookup,
-                                                                                                    permissions_lookup)).create()
+            resclass_type = resclass_name_2_type[resource.restype]
+            properties = resource.get_propvals(res_iri_lookup, permissions_lookup)
+            resclass_instance: ResourceInstance = resclass_type(
+                con=con,
+                label=resource.label,
+                iri=resource_iri,
+                permissions=permissions_tmp,
+                bitstream=resource_bitstream,
+                values=properties
+            )
+            resclass_instance = resclass_instance.create()
         except BaseError as err:
-            print(
-                f"ERROR while trying to create resource '{resource.label}' ({resource.id}). The error message was: {err.message}")
+            print(f"ERROR while trying to create resource '{resource.label}' ({resource.id}). "
+                  f"The error message was: {err.message}")
             failed_uploads.append(resource.id)
             continue
-
         except Exception as exception:
-            print(
-                f"ERROR while trying to create resource '{resource.label}' ({resource.id}). The error message was: {exception}")
+            print(f"EXCEPTION while trying to create resource '{resource.label}' ({resource.id}). "
+                  f"The exception message was: {exception}")
             failed_uploads.append(resource.id)
             continue
 
-        res_iri_lookup[resource.id] = instance.iri
-        print(f"Created resource '{instance.label}' ({resource.id}) with IRI '{instance.iri}'")
+        res_iri_lookup[resource.id] = resclass_instance.iri
+        print(f"Created resource '{resclass_instance.label}' ({resource.id}) with IRI '{resclass_instance.iri}'")
+
+    # update the resources with the stashed XML texts
+    if len(stashed_xml_texts) > 0:
+        print('Update the stashed XML texts...')
+    for resource, link_props in stashed_xml_texts.items():
+        print(f'Update XML text(s) of resource "{resource.id}"...')
+        res_iri = res_iri_lookup[resource.id]
+        try:
+            update_xml_texts(
+                resource=resource,
+                res_iri=res_iri,
+                link_props=link_props,
+                res_iri_lookup=res_iri_lookup,
+                con=con,
+                verbose=verbose
+            )
+        except BaseError as err:
+            print(f'BaseError while updating an XML text of resource "{resource.id}": {err.message}')
```

Comment:
The user should not care about what sort of error it was. He just needs to know that there was an error. So, I would use "ERROR" (see other error messages for the format / formulation) instead of "BaseError".

Reply:
I thought this difference would help us when debugging. If I get an error message, I like to copy it and search it in the source code. For this purpose, it helps to have a specific error message.

Reply:
I see, but "BaseError" doesn't tell you much more than "error", does it? The interesting information is in the error message itself. And, the general problem is that we don't have logging in dsp-tools. That would be the place to write out more information. The end user should IMO not be bothered with technical details.